### ISSUE TITLE
UBSan builds fail if Spicy is enabled

### ISSUE THREAD

**Issue by timwoj** [zeek/zeek#3670]
We disable spicy on CI, probably for this reason, but I figured I should file it anyways.

```
[1821/1850] Compiling WebSocket analyzer
FAILED: src/analyzer/protocol/websocket/websocket___linker__.cc src/analyzer/protocol/websocket/websocket_spicy_init.cc src/analyzer/protocol/websocket/websocket_spicy_hooks_WebSocket.cc src/analyzer/protocol/websocket/websocket_WebSocket.cc /Users/tim/Desktop/projects/zeek/build/src/analyzer/protocol/websocket/websocket___linker__.cc /Users/tim/Desktop/projects/zeek/build/src/analyzer/protocol/websocket/websocket_spicy_init.cc /Users/tim/Desktop/projects/zeek/build/src/analyzer/protocol/websocket/websocket_spicy_hooks_WebSocket.cc /Users/tim/Desktop/projects/zeek/build/src/analyzer/protocol/websocket/websocket_WebSocket.cc 
cd /Users/tim/Desktop/projects/zeek/src/analyzer/protocol/websocket && /opt/homebrew/Cellar/cmake/3.28.1/bin/cmake -E env ZEEK_SPICY_LIBRARY_PATH=/Users/tim/Desktop/projects/zeek/scripts/spicy ASAN_OPTIONS=detect_leaks=0:detect_container_overflow=0:detect_leaks=0 /Users/tim/Desktop/projects/zeek/build/src/spicy/spicyz/spicyz -L /Users/tim/Desktop/projects/zeek/auxil/spicy/hilti/lib -L /Users/tim/Desktop/projects/zeek/auxil/spicy/spicy/lib -x /Users/tim/Desktop/projects/zeek/build/src/analyzer/protocol/websocket/websocket websocket.spicy websocket.evt unmask.cc
/Users/tim/Desktop/projects/zeek/auxil/spicy/hilti/toolchain/src/ast/scope-lookup.cc:14:33: runtime error: member call on null pointer of type 'hilti::Scope'
    #0 0x110c6de58 in hilti::scope::detail::lookupID(hilti::ID const&, hilti::Node const*) scope-lookup.cc:14
    #1 0x10891cc2c in hilti::rt::Result<std::__1::pair<hilti::declaration::Type*, hilti::ID>> hilti::scope::lookupID<hilti::declaration::Type>(hilti::ID, hilti::Node*, std::__1::basic_string_view<char, std::__1::char_traits<char>> const&) scope-lookup.h:40
    #2 0x1117096f0 in (anonymous namespace)::VisitorPass1::operator()(hilti::type::Name*) resolver.cc:59
    #3 0x110cc8a14 in hilti::type::Name::dispatch(hilti::visitor::Dispatcher&) name.h:66
    #4 0x11170ae40 in hilti::visitor::Visitor<(hilti::visitor::Order)1, hilti::visitor::Dispatcher>::dispatch(hilti::Node*) visitor.h:183
    #5 0x111707914 in auto hilti::visitor::visit<(anonymous namespace)::VisitorPass1&, hilti::Node>((anonymous namespace)::VisitorPass1&, hilti::Node*, std::__1::basic_string_view<char, std::__1::basic_string_view::char_traits<char>>) visitor.h:334
    #6 0x111706e5c in hilti::detail::resolver::resolve(hilti::Builder*, hilti::Node*) resolver.cc:1619
    #7 0x11164d034 in hilti::detail::createHiltiPlugin()::$_9::operator()(hilti::Builder*, hilti::Node*) const plugin.cc:87
    #8 0x11164d000 in decltype(std::declval<hilti::detail::createHiltiPlugin()::$_9&>()(std::declval<hilti::Builder*>(), std::declval<hilti::Node*>())) std::__1::__invoke[abi:ue170006]<hilti::detail::createHiltiPlugin()::$_9&, hilti::Builder*, hilti::Node*>(hilti::detail::createHiltiPlugin()::$_9&, hilti::Builder*&&, hilti::Node*&&) invoke.h:340
    #9 0x11164ce80 in bool std::__1::__invoke_void_return_wrapper<bool, false>::__call[abi:ue170006]<hilti::detail::createHiltiPlugin()::$_9&, hilti::Builder*, hilti::Node*>(hilti::detail::createHiltiPlugin()::$_9&, hilti::Builder*&&, hilti::Node*&&) invoke.h:407
    #10 0x11164cdd0 in std::__1::__function::__alloc_func<hilti::detail::createHiltiPlugin()::$_9, std::__1::allocator<hilti::detail::createHiltiPlugin()::$_9>, bool (hilti::Builder*, hilti::Node*)>::operator()[abi:ue170006](hilti::Builder*&&, hilti::Node*&&) function.h:193
    #11 0x111647110 in std::__1::__function::__func<hilti::detail::createHiltiPlugin()::$_9, std::__1::allocator<hilti::detail::createHiltiPlugin()::$_9>, bool (hilti::Builder*, hilti::Node*)>::operator()(hilti::Builder*&&, hilti::Node*&&) function.h:364
    #12 0x10896f0d4 in std::__1::__function::__value_func<bool (hilti::Builder*, hilti::Node*)>::operator()[abi:ue170006](hilti::Builder*&&, hilti::Node*&&) const function.h:518
    #13 0x1088e0884 in std::__1::function<bool (hilti::Builder*, hilti::Node*)>::operator()(hilti::Builder*, hilti::Node*) const function.h:1169
    #14 0x1088e01d8 in spicy::detail::resolver::resolve(hilti::ExtendedBuilderTemplate<spicy::BuilderBase>*, hilti::Node*) resolver.cc:628
    #15 0x1088b3b50 in spicy::detail::createSpicyPlugin()::$_6::operator()(hilti::Builder*, hilti::Node*) const plugin.cc:70
    #16 0x1088b3a60 in decltype(std::declval<spicy::detail::createSpicyPlugin()::$_6&>()(std::declval<hilti::Builder*>(), std::declval<hilti::Node*>())) std::__1::__invoke[abi:ue170006]<spicy::detail::createSpicyPlugin()::$_6&, hilti::Builder*, hilti::Node*>(spicy::detail::createSpicyPlugin()::$_6&, hilti::Builder*&&, hilti::Node*&&) invoke.h:340
    #17 0x1088b38e0 in bool std::__1::__invoke_void_return_wrapper<bool, false>::__call[abi:ue170006]<spicy::detail::createSpicyPlugin()::$_6&, hilti::Builder*, hilti::Node*>(spicy::detail::createSpicyPlugin()::$_6&, hilti::Builder*&&, hilti::Node*&&) invoke.h:407
    #18 0x1088b3830 in std::__1::__function::__alloc_func<spicy::detail::createSpicyPlugin()::$_6, std::__1::allocator<spicy::detail::createSpicyPlugin()::$_6>, bool (hilti::Builder*, hilti::Node*)>::operator()[abi:ue170006](hilti::Builder*&&, hilti::Node*&&) function.h:193
    #19 0x1088adb70 in std::__1::__function::__func<spicy::detail::createSpicyPlugin()::$_6, std::__1::allocator<spicy::detail::createSpicyPlugin()::$_6>, bool (hilti::Builder*, hilti::Node*)>::operator()(hilti::Builder*&&, hilti::Node*&&) function.h:364
    #20 0x10ff6f3ec in std::__1::__function::__value_func<bool (hilti::Builder*, hilti::Node*)>::operator()[abi:ue170006](hilti::Builder*&&, hilti::Node*&&) const function.h:518
    #21 0x10ff6ef5c in std::__1::function<bool (hilti::Builder*, hilti::Node*)>::operator()(hilti::Builder*, hilti::Node*) const function.h:1169
    #22 0x10fe18b64 in hilti::rt::Result<hilti::rt::Nothing> _runHook<std::__1::optional<std::__1::function<bool (hilti::Builder*, hilti::Node*)>> hilti::Plugin::*, hilti::Builder*, hilti::ASTRoot*>(bool*, hilti::Plugin const&, std::__1::optional<std::__1::function<bool (hilti::Builder*, hilti::Node*)>> hilti::Plugin::*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, hilti::Builder* const&, hilti::ASTRoot* const&) ast-context.cc:321
    #23 0x10fe18548 in hilti::ASTContext::_resolveRoot(bool*, hilti::Builder*, hilti::Plugin const&) ast-context.cc:456
    #24 0x10fe0f67c in hilti::ASTContext::_resolve(hilti::Builder*, hilti::Plugin const&) ast-context.cc:483
    #25 0x10fe0cb48 in hilti::ASTContext::processAST(hilti::Builder*, hilti::Driver*) ast-context.cc:348
    #26 0x111442388 in hilti::Driver::compileUnits() driver.cc:649
    #27 0x11144643c in hilti::Driver::compile() driver.cc:724
    #28 0x1048491d4 in zeek::spicy::Driver::compile() driver.cc:218
    #29 0x104c2281c in main main.cc:274
    #30 0x1898260dc  (<unknown module>)
```

@bbannier @rsmmr Should I refile this in the spicy repo instead? To repro, I ran this as configure:

```
ZEEK_TAILORED_UB_CHECKS=1 UBSAN_OPTIONS=print_stacktrace=1 NO_OPTIMIZATIONS=1 ./configure --build-type=debug --generator=Ninja --sanitizers=address,undefined
```

**Comment by bbannier** [comment]
@timwoj, let's leave this in this repo since getting it enabled in Zeek would be the ultimate goal.

**Comment by awelzel** [comment]
Fixed through zeek/spicy#1817.
